src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  let oldYearFormat = '';
+  const inputDateParts = date.split(oldSeparator);
+  const newDateParts = [];
+
+  if (fromFormat.includes('YY')) {
+    oldYearFormat = 'YY';
+  } else if (fromFormat.includes('YYYY')) {
+    oldYearFormat = 'YYYY';
+  }
+
+  for (let i = 0; i < 3; i++) {
+    const newFormat = toFormat[i];
+    let value = '';
+
+    /**
+     * Define the value of the given date part.
+     * It remains unchanged if the format is the same.
+     * No need to check if new format is equal to 'YY' or 'YYYY' as this is
+     * done while setting the value for oldYearFormat.
+     *
+     * In other case call the 'convertYear' function to reformat the value.
+     */
+    if (['DD', 'MM'].includes(newFormat) || newFormat === oldYearFormat) {
+      const valueIndex = fromFormat.indexOf(newFormat);
+
+      value = inputDateParts[valueIndex];
```

Comment:
The condition in this if statement is checking if 'newFormat' is 'DD', 'MM', or the same as 'oldYearFormat'. This logic might not be necessary since you're iterating over 'toFormat' and should map each part directly by index, considering that 'fromFormat' and 'toFormat' should have the same parts in different order.